### PR TITLE
create frogger-game

### PR DESCRIPTION
**Pull request by Pozzitive11**
### Frogger game

[Demo](https://pozzitive11.github.io/frogger-game-/) |
[Code base](https://github.com/Pozzitive11/frogger-game-)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/pozzitive11/frogger-game/app.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    this.y = y;
+    this.speed = speed;
+    this.enemyCenter = 35;
+  }
+
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
+
+  collision() {
```
Comment (OleksiyRudenko) [inline submissions/pozzitive11/frogger-game/app.js:25]: BY convention method/function names should start with a verb.

File: submissions/pozzitive11/frogger-game/app.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  update() {}
+}
+
+const player = new Player();
+const enemy = new Enemy();
+
+const allEnemies = [
+  new Enemy(tileSize.height * 3 - enemy.enemyCenter, 90),
+  new Enemy(tileSize.height * 2 - enemy.enemyCenter, 150),
+  new Enemy(tileSize.height * 1 - enemy.enemyCenter, 100),
```
Comment (OleksiyRudenko) [inline submissions/pozzitive11/frogger-game/app.js:99]: What are these magic numbers - 90, 150, 100?
Let the code explain them.

File: submissions/pozzitive11/frogger-game/app.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+  resetToStartPosition() {
+    if (this.y < boardSize.top) {
+      this.moveToStartPosition();
+    }
+  }
+
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
```
Comment (OleksiyRudenko) [inline submissions/pozzitive11/frogger-game/app.js:63]: ENemy class has exactly the same method. Let's make this [DRY](https://www.digitalocean.com/community/tutorials/what-is-dry-development). Consider having a base class for both Player and Enemy, that would handle all shared properties and behaviour.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Pozzitive11 well done!
Let's use the opportunity to improve the code and be ready for a massive expansion of game characters to be added by our peer developers.

File: submissions/pozzitive11/frogger-game/app.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+};
+
+class gameObjectRender {
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
+}
+class Enemy extends gameObjectRender {
+  constructor(y, speed, sprite) {
+    super(y, sprite);
```
Comment (OleksiyRudenko) [inline submissions/pozzitive11/frogger-game/app.js:26]: Super class doesn't have a constructor. This operation is effectively void. Same is applicable to the `Player` class. However the idea to have similar properties and similar behaviour in a base class is great.
Can you fix this and implement shared properties and methods via the base class?

File: submissions/pozzitive11/frogger-game/app.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+};
+
+const boardSize = {
+  top: 0,
+  right: tileSize.width * 5,
+  bottom: tileSize.height * 6,
+  left: 0,
+};
+
+class gameObjectRender {
```
Comment (OleksiyRudenko) [inline submissions/pozzitive11/frogger-game/app.js:19]: Think of class hierarchy as of bilogical taxonomy. Humans inherit features from apes, apes from mammals, mammals from vertebraes, and so on.
In this fictional game world bugs and player have a common fictional predecessor whose name could be saying something to the developers that may want to contribute to your game. May be a both bugs and player are instances of a character?

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Pozzitive11 your changes feel unfinished. Let's polish it.

**Comment by Pozzitive11** [comment]
sorry for making the same mistakes 🥲

**Review by OleksiyRudenko — APPROVED** [review]
@Pozzitive11 great job!

**Pull request merged**